### ISSUE TITLE
No support of long paths

### ISSUE THREAD

**Issue by AlttiRi** [arsenetar/dupeguru#893]
**Describe the bug**
The program does not support long paths. It just ignores files with a path is over ~255 characters.

**To Reproduce**
- Download this zip: [file_structure.zip](https://github.com/arsenetar/dupeguru/files/6511722/file_structure.zip)
- Unpack it
- Try to find duplicates.

It finds only the 2 files with short paths.

**Expected behavior**
It should to find all duplicate files regardless the length of their paths.

There are in the zip:
6 original files _(3 unique images and 3 unique txt files)_,19 total files.

**Screenshots**
![Screenshot](https://user-images.githubusercontent.com/16310547/118889519-e30a2500-b905-11eb-87d1-80878be5ebf4.png)

**Desktop (please complete the following information):**
 - Windows 10
 - 4.1.0

**Comment by glubsy** [comment]
This is normal, and won't be fixed. You cannot go beyond the limits: 255 bytes for a filename on GNU/Linux, 255 characters for a filename in NTFS. See [reference](https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits).

You mention "file path", but it seems you actually meant "file name length". You are not even hitting the path length limit here anyway, except for the directory name that is limited to 255 (a directory is also a file).

**Comment by AlttiRi** [comment]
**A talk especially about the total path' length.** In my zip there is no file with _path chunk_ what contains more that 255 characters, because it impossible to do for both Windows and Linux.

But the total length of each paths of the files in the archive is more than 255 characters and the program does not work with it, but should.

Windows 10 “Enable NTFS long paths policy”
https://superuser.com/a/1119980/1034786

In windows the max length of the path can be ~**32767**  characters.
https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation

> See [reference](https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits).

> NTFS, (Maximum pathname length)
> 32,767 Unicode characters with each path component (directory or filename) up to 255 characters long

## TL;DR 
_This_ program does not work correctly with files in this archive.
[file_structure.zip](https://github.com/arsenetar/dupeguru/files/6511722/file_structure.zip) _(Not the best name of the archive_)

While the other programs work well.

**Comment by AlttiRi** [comment]
Here is the correct result:

![image](https://user-images.githubusercontent.com/16310547/118910291-e5cc4080-b92c-11eb-9f10-6dcea137f73a.png)

**Comment by AlttiRi** [comment]
6 original files and 19 total files.

3 unique images and 3 unique txt files.

**Comment by arsenetar** [comment]
@AlttiRi If I use the zip you provided, I am getting the correct results when doing a contents or filename scan without having the ignore small files option selected (text files are ignored otherwise).  I am running the 4.1.1 build but there should not be any changes that would impact this between this version and 4.1.0.

**Comment by arsenetar** [comment]
@AlttiRi If you enable the debug log and provide the output of the log after a scan that might help provide some insight as to why you are getting a different result.

**Comment by AlttiRi** [comment]
`
2021-02-20 17:34:18,270 - WARNING - Error while loading C:/Users/Username/AppData/Local/Hardcoded Software/dupeGuru\exclude_list.xml: [Errno 2] No such file or directory: 'C:/Users/Username/AppData/Local/Hardcoded Software/dupeGuru\\exclude_list.xml'
`

UPD: Hm, `2021-02-20`. It's probably the log of something other.

**Comment by AlttiRi** [comment]
Well, with 4.1.1 it works.

![image](https://user-images.githubusercontent.com/16310547/118917569-d9021980-b939-11eb-86ee-81e517ec3d4b.png)

I checked the change log before creating this issue, I did not see something that could be affect on this.
But it works now.

---
https://docs.python.org/3/using/windows.html#removing-the-max-path-limitation
> Changed in version 3.6: Support for long paths was enabled in Python.

Although 4.1.0 also was built with Python 3.8.
